lib/HALTarget/IMU.cpp

```diff
@@ -0,0 +1,211 @@
+        break;
+    default:
+        ;
+    }
+    return isInitSuccessful;
+}
+
+void IMU::enableDefault()
+{
+    m_imu.enableDefault();
+}
+
+void IMU::configureForTurnSensing()
+{
+    m_imu.configureForTurnSensing();
+}
+
+void IMU::readAcc()
+{
+    while(!m_imu.accDataReady()){}
+    m_imu.readAcc();
+    m_accelerometerValues.x = m_imu.a.x;
+    m_accelerometerValues.y = m_imu.a.y;
+    m_accelerometerValues.z = m_imu.a.z;
+}
+
+void IMU::readGyro()
+{
+    while(!m_imu.gyroDataReady()){}
+    m_imu.readGyro();
+    m_gyroValues.x = m_imu.g.x;
+    m_gyroValues.y = m_imu.g.y;
+    m_gyroValues.z = m_imu.g.z;
+}
+
+void IMU::readMag()
+{
+    while(!m_imu.magDataReady()){}
+    m_imu.readMag();
+    m_magnetometerValues.x = m_imu.m.x;
+    m_magnetometerValues.y = m_imu.m.y;
+    m_magnetometerValues.z = m_imu.m.z;
+}
+
+bool IMU::accDataReady()
+{
+    return m_imu.accDataReady();
+}
+
+bool IMU::gyroDataReady()
+{
+    return m_imu.gyroDataReady();
+}
+
+bool IMU::magDataReady()
+{
+    return m_imu.magDataReady();
+}
+
+void IMU::getAccelerationValues(int16_t* accelerationValues)
+{
+    accelerationValues[AXIS_INDEX_X] = m_accelerometerValues.x - m_rawAccelerometerOffsetX;
+    accelerationValues[AXIS_INDEX_Y] = m_accelerometerValues.y - m_rawAccelerometerOffsetY;
+    accelerationValues[AXIS_INDEX_Z] = m_accelerometerValues.z - m_rawAccelerometerOffsetZ;
+}
+
+void IMU::getTurnRates(int16_t* turnRates)
+{
+    turnRates[AXIS_INDEX_X] = m_gyroValues.x - m_rawGyroOffsetX;
+    turnRates[AXIS_INDEX_Y] = m_gyroValues.y - m_rawGyroOffsetY;
+    turnRates[AXIS_INDEX_Z] = m_gyroValues.z - m_rawGyroOffsetZ;
+}
+
+void IMU::getMagnetometerValues(int16_t* magnetometerValues)
+{
+    magnetometerValues[AXIS_INDEX_X] = m_magnetometerValues.x;
+    magnetometerValues[AXIS_INDEX_Y] = m_magnetometerValues.y;
+    magnetometerValues[AXIS_INDEX_Z] = m_magnetometerValues.z;
+}
+
+// TD076	Make sure that it is necessary to read and calibrate all sensor values, if only 1 of the gyro values and 2 of the acceleration values are being used.
+void IMU::calibrate()
+{    
+    /* Define how many measurements should be made for calibration. */
+    const uint8_t NUMBER_OF_MEASUREMENTS      = 10; 
+
+    /* Calibration takes place while the robot doesn't move. Therefore the Acceleration and turn values are near 0 and int16_t are enough. */
+    int16_t sumOfRawAccelValuesX      = 0;
+    int16_t sumOfRawAccelValuesY      = 0;
+    int16_t sumOfRawAccelValuesZ      = 0;
+    
+    int16_t sumOfRawGyroValuesX       = 0;
+    int16_t sumOfRawGyroValuesY       = 0;
+    int16_t sumOfRawGyroValuesZ       = 0;
+
+    for (uint8_t measurementIdx=0; measurementIdx<NUMBER_OF_MEASUREMENTS; ++measurementIdx)
+    {
+        while(!m_imu.accDataReady()){}
+        m_imu.readAcc();
+        sumOfRawAccelValuesX += m_imu.a.x;
+        sumOfRawAccelValuesY += m_imu.a.y;
+        sumOfRawAccelValuesZ += m_imu.a.z;
+        
+        while(!m_imu.gyroDataReady()){}
```

Comment:
Please use multiple rows & add comment that nothing is to be done while waiting.